components/plugin_marketplace/marketplace_item/marketplace_item_plugin/marketplace_item_plugin.tsx

```diff
@@ -0,0 +1,378 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import classNames from 'classnames';
+import semver from 'semver';
+
+import {FormattedMessage} from 'react-intl';
+
+import {Link} from 'react-router-dom';
+
+import type {MarketplaceLabel} from 'mattermost-redux/types/marketplace';
+
+import MarketplaceItem from '../marketplace_item';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import ConfirmModal from 'components/confirm_modal';
+import LoadingWrapper from 'components/widgets/loading/loading_wrapper';
+
+import {localizeMessage} from 'utils/utils';
+
+type UpdateVersionProps = {
+    version: string;
+    releaseNotesUrl?: string;
+};
+
+// UpdateVersion renders the version text in the update details, linking out to release notes if available.
+export const UpdateVersion = ({version, releaseNotesUrl}: UpdateVersionProps): JSX.Element => {
+    if (!releaseNotesUrl) {
+        return (
+            <span>
+                {version}
+            </span>
+        );
+    }
+
+    return (
+        <a
+            target='_blank'
+            rel='noopener noreferrer'
+            href={releaseNotesUrl}
+        >
+            {version}
+        </a>
+    );
+};
+
+export type UpdateDetailsProps = {
+    version: string;
+    releaseNotesUrl?: string;
+    installedVersion?: string;
+    isInstalling: boolean;
+    onUpdate: (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+};
+
+// UpdateDetails renders an inline update prompt for plugins, when available.
+export const UpdateDetails = ({version, releaseNotesUrl, installedVersion, isInstalling, onUpdate}: UpdateDetailsProps): JSX.Element | null => {
+    if (!installedVersion || isInstalling) {
+        return null;
+    }
+
+    let isUpdate = false;
+    try {
+        isUpdate = semver.gt(version, installedVersion);
+    } catch (e) {
+        // If we fail to parse the version, assume not an update;
+    }
+
+    if (!isUpdate) {
+        return null;
+    }
+
+    return (
+        <div className={classNames('update')}>
+            <FormattedMessage
+                id='marketplace_modal.list.update_available'
+                defaultMessage='Update available:'
+            />
+            {' '}
+            <UpdateVersion
+                version={version}
+                releaseNotesUrl={releaseNotesUrl}
+            />
+            {' - '}
+            <b>
```

Comment:
Good point. For context, this line existed before this PR and only appears in the diff because of the TS migration. 2/5 to not touch the functionality of that code in this PR.